text/0042-separate-context-propagation.md

```diff
@@ -0,0 +1,213 @@
+In order for the application layer to function correctly, Propagation choices must be syncronized between all processes in the distributed system, and multiple applications must be able to inject and extract their context into the same request. To meet these requirements, the Propagation API provides a function which registers a set of propagators, which will all be executed in order when the future calls to inject and extract are made. A canonical propagator consists of an inject and an extract function.
+
+### Optional: Global Propagators
+It is often convenient to create a chain of propagators during program initialization, and then access these combined propagators later in the program. To facilitate this, global injectors and extractors are optionally available. However, there is no requirement to use this feature.
+
+**SetHTTPInjector(injector)**  
+To update the global injector, the Propagation API provides a function which takes an injector.
+
+**GetHTTPInjector() -> injector**  
+To access the global injector, the Propagation API provides a function which returns an injector.
+
+**SetHTTPExtractor(extractor)**  
+To update the global extractor, the Propagation API provides a function which takes an injector.
+
+**GetHTTPExtractor() -> extractor**  
+To access the global extractor, the Propagation API provides a function which returns an extractor.
+
+# Internal details
+
+![drawing](img/context_propagation_details.png)
+
+## Context details
+OpenTelemetry currently implements three context types of context propagation.
+
+**Span Context -** The serializable portion of a span, which is injected and extracted. The readable attributes are defined to match those found in the  W3C **traceparent** header. 
+
+**Correlation Context -** Correlation Context contains a map of labels and values, to be shared between metrics and traces. This allows observability data to be indexed and dimensionalized in a variety of ways. Note that correlations can quickly add overhead when propagated in-band. But because this data is write-only, it may be possible to optimize how it is transmitted.
+
+**Baggage Context -** Transaction-level application data, meant to be shared with downstream components. This data is readable, and must be propagated in-band. Because of this, Baggage should be used sparingly, to avoid ballooning the size of RPC requests.
+
+Note that when possible, OpenTelemetry APIs calls are given access to the entire context object, and not a specific context type.
+
+
+## Context Management and in-process propagation
+
+In order for Context to function, it must always remain bound to the execution of code it represents. By default, this means that the programmer must pass a Context down the call stack as a function parameter. However, many languages provide automated context management facilities, such as thread locals. OpenTelemetry should leverage these facilities when available, in order to provide automatic context management.
```

Comment:
Not for setting the active span in the current context, but when moving work from one thread to another, or context switching in async/nonblocking runtimes, this needs to be addressed. 

In general, I'm concerned that there will be important details which present themselves better when we attempt to implement this. We've gotten a bit of a pass so far because we did so much work in java at the very beginning of this process, before RFCs, etc. FWIW, in OpenTracing, we required a working implementation before adding an committing a change to the spec.